torch/csrc/jit/codegen/cuda/lower_alias_memory.cpp

```diff
@@ -425,37 +557,45 @@ using AllocationInfoList = std::vector<AllocationInfoPtr>;
 //!   Outer interval marks the beginning of the loop of first write and end of
 //!   the loop of last read, both at the same loop level as the buffer
 //!   allocation.
-class BufferUseDefInfo {
```

Comment:
This class traverses the given expression vector twice. The first pass creates a meta-data property for each scope, i.e., `kir::ForLoop`. The second pass determines the live range of each tensor allocation using the scope properties built in the first pass.

In this PR, I separated out the first pass as `ScopeMap`. The remaining part, now called `AllocationInfoMap`, only does the second pass.